test/test.js

```diff
@@ -213,6 +213,43 @@ describe('Routes', () => {
       ])
       assert.strictEqual(measurementRow.participant_address, participantAddress)
     })
+
+    it('handles date fields set to null', async () => {
+      await client.query('DELETE FROM measurements')
+
+      const measurement = {
+        cid: 'bafytest',
+        providerAddress: '/dns4/localhost/tcp/8080',
+        protocol: 'graphsync',
+        sparkVersion: '1.2.3',
+        zinniaVersion: '2.3.4',
+        participantAddress,
+        startAt: new Date(),
+        statusCode: undefined,
+        firstByteAt: null,
+        endAt: null
+      }
+
+      const createRequest = await fetch(`${spark}/measurements`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(measurement)
+      })
+      await assertResponseStatus(createRequest, 200)
+      const { id } = await createRequest.json()
+
+      const { rows: [measurementRow] } = await client.query(`
+          SELECT *
+          FROM measurements
+          WHERE id = $1
+        `, [
+        id
+      ])
+
+      assert.strictEqual(measurementRow.status_code, null)
```

Comment:
`status_code` is NULLABLE in the database. When the measurement does not specify the status code at all (value is `undefined`), the roundtrip through the database converts the value to `null`.

This PR does not change the existing behaviour, just adds a test to document it.

🤷🏻‍♂️